src/OrleansRuntime/Hosting/SiloWrapper.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Hosting
+{
+    internal class SiloWrapper : ISilo
+    {
+        private readonly Silo silo;
+
+        public SiloWrapper(Silo silo, IServiceProvider services)
+        {
+            this.Services = services;
+            this.silo = silo;
+            this.Stopped = silo.SiloTerminated;
+        }
+
+        /// <inheritdoc />
+        public IServiceProvider Services { get; }
+
+        /// <inheritdoc />
+        public Task Stopped { get; }
+
+        /// <inheritdoc />
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            this.silo.Start();
+            return Task.CompletedTask;
```

Comment:
Yes, true, I thought about the same thing... it's not entirely clear in the spec in aspnet/Hosting#1163.
Perhaps @davidfowl can clarify what's the expected behavior when the token is cancelled before `StartAsync` finishes?
I believe that `StopAsync` case is clear: it should ungracefully stop when the token is canceled, as the process can be terminated immediately.